screenpipe-vision/src/core.rs

```diff
@@ -91,15 +99,78 @@ pub async fn continuous_capture(
+                            let previous_text_json_len: usize = previous_text_json.as_ref().map_or(0, |v: &Vec<HashMap<String, String>>| v.iter().map(|s| s.len()).sum());
 
                             debug!("JSON length Current: {} Previous: {}", current_text_json_len, previous_text_json_len);
 
+                            // Compare current and previous text_json to create new_text_json
+                            let mut new_text_json = Vec::new();
+                            if let Some(prev_json) = &*previous_text_json {
+                                for current_record in &current_text_json {
+                                    let confidence: f64 = current_record["confidence"].parse().unwrap_or(0.0);
+                                    if confidence > 60.0 {
+                                        let is_new = prev_json.iter().all(|prev_record| {
+                                            let distance = levenshtein(&current_record["text"], &prev_record["text"]);
+                                            let threshold = (prev_record["text"].len() as f64 * 0.1).ceil() as usize;
+                                            distance > threshold
+                                        });
+                                        if is_new {
+                                            new_text_json.push(current_record.clone());
+                                        }
+                                    }
+                                }
+                            } else {
+                                new_text_json = current_text_json.iter()
+                                    .filter(|record| record["confidence"].parse::<f64>().unwrap_or(0.0) > 60.0)
+                                    .cloned()
+                                    .collect();
+                            }
+
+                            // Save text files
+                            let id = frame_number; // Use frame_number as the incremental ID
+                            debug!("Saving text files for frame {}", frame_number);
+
+                            // Extract raw text lines from new_text_json
+                            let new_text_lines: Vec<String> = new_text_json.iter().map(|record| {
+                                record.get("text").cloned().unwrap_or_default()
+                            }).collect();
+
+                            // Extract raw text lines from current_text_json
+                            let current_text_lines: Vec<String> = current_text_json.iter().map(|record| {
+                                record.get("text").cloned().unwrap_or_default()
+                            }).collect();
+
+                            // Save new text lines to file
+                            let new_text_file_path = format!("text_json/new_text_{}.txt", id);
+                            let mut new_text_file = File::create(&new_text_file_path).unwrap();
```

Comment:
running this for a few hours and lets merge then

commented out this, assuming it was a dev/debug part:

```rs
// Extract raw text lines from new_text_json
                            // let new_text_lines: Vec<String> = new_text_json.iter().map(|record| {
                            //     record.get("text").cloned().unwrap_or_default()
                            // }).collect();

                            // // Extract raw text lines from current_text_json
                            // let current_text_lines: Vec<String> = current_text_json.iter().map(|record| {
                            //     record.get("text").cloned().unwrap_or_default()
                            // }).collect();

                            // Save new text lines to file
                            // let new_text_file_path = format!("text_json/new_text_{}.txt", id);
                            // let mut new_text_file = File::create(&new_text_file_path).unwrap();
                            // for line in new_text_lines {
                            //     writeln!(new_text_file, "{}", line).unwrap();
                            // }

                            // Save current text lines to file
                            // let current_text_file_path = format!("text_json/current_text_{}.txt", id);
                            // let mut current_text_file = File::create(&current_text_file_path).unwrap();
                            // for line in current_text_lines {
                            //     writeln!(current_text_file, "{}", line).unwrap();
                            // }

                            // Save previous text lines to file if available
                            // if let Some(prev_json) = &*previous_text_json {
                            //     let prev_text_lines: Vec<String> = prev_json.iter().map(|record| {
                            //         record.get("text").cloned().unwrap_or_default()
                            //     }).collect();
                            //     let prev_text_file_path = format!("text_json/previous_text_{}.txt", id);
                            //     let mut prev_text_file = File::create(&prev_text_file_path).unwrap();
                            //     for line in prev_text_lines {
                            //         writeln!(prev_text_file, "{}", line).unwrap();
                            //     }
                            // }
```